composer/trainer/trainer.py

```diff
@@ -47,6 +47,35 @@
 
 log = logging.getLogger(__name__)
 
+def _dynamic_microbatch_wrapper(func: Callable) -> Callable:
+    """Wraps function to catch CUDA Out of Memory Errors and adaptively change microbatch
+    size if enabled to miaximize GPU usage.
+    """
+    def wrapper(*args, **kwargs):
+        self = args[0]
```

Comment:
Do you know if there's a better way to handle these types? Can `_dynamic_microbatch_wrapper` be defined specifically as a wrapper for methods of a given type? Maybe something in `functools` can help you?

Reply:
Can this just be written as `def wrapper(self, *args, **kwargs)`? Having `self = args[0]` without any safety checks is not great style